koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -773,7 +792,8 @@ def createContainer(self, src=None, target_info=None, arches=None,
         except OsbsValidationException as exc:
             raise ContainerError('OSBS validation exception: {0}'.format(exc))
 
-        return self.handle_build_response(self.osbs().get_build_name(build_response))
+        return self.handle_build_response(self.osbs().get_build_name(build_response),
+                                          platforms=arches)
```

Comment:
What is the relationship between the `arches` here and the platforms that actually end up being used in atomic-reactor (determined by the check_and_set_platforms plugin)?

If these arches are not guaranteed to be a superset of the platforms actually used, that would be a problem. For example, let's say the build will run s390x tasks, but because s390x was not in arches, koji-c will not check for that platform. That means s390x logs will go to the noarch log.

Reply:
It kind of looks like there _is_ a nearly impossible corner case where this would be a problem. If someone uses the `arch_override` option and specifies a non-canonical name for one of the platforms supported by the build tag (e.g. [ia32e for x86_64](https://pagure.io/koji/blob/77aec93b98b6bfe2b870e41c5c28949d2ba06412/f/koji/__init__.py#_1213)), this non-canonical name would be passed to atomic-reactor.
https://github.com/containerbuildsystem/koji-containerbuild/blob/0667b30ff9cf63cc285d23905ee77b9c72016497/koji_containerbuild/plugins/builder_containerbuild.py#L785-L800

But that would cause far bigger issues than just misplacing the logs, plus arch_override is only allowed for scratch and isolated builds anyway.

In all cases except this (IMO irrelevant) one, it appears the arches _are_ a superset.

Reply:
yes, except for `arch_override`, `arches` are superset of `platforms` we get from the `check_and_set_platforms` plugin.